### ISSUE TITLE
Compilation time

### ISSUE THREAD

**Issue by loewenstein** [soundcloud/haskell-kubernetes#7]
I noticed that (from scratch [1]) compilation time for this repo is ~20 minutes.

I am not sure what is the main culprit. I could reduce a bit by excluding the `servant-server` dependency. 
However that was not yet satisfactory.

Any idea how we could further reduce compilation time?

[1] 
```
rm -rf ~/.stack
rm -rf .stack-work
stack setup
stack build
```

**Comment by aaronlevin** [comment]
@loewenstein the main culprit is the `client`. Probably best to just not generate the client, or somehow make that optional / explicit.

**Comment by aaronlevin** [comment]
see: https://github.com/soundcloud/haskell-kubernetes/blob/master/haskell-kubernetes.cabal#L171

what's weird is that the `server` is commented out. i have no idea why that is, though I am surely to blame.

**Comment by aaronlevin** [comment]
(tbh I only remember it being the client, I may have been wrong)

**Comment by loewenstein** [comment]
Mmh, removing the client I am left with ~8 minutes. Still not very fast I would say.

Could that be related to the use of TH in all the model files?
Is it really worth generating TH code in swagger-codegen. Aeson and Lenses TH is all about conciseness, isn't it? But then, it is generated code anyway.

What do you think?

**Comment by aaronlevin** [comment]
@loewenstein we could update the code gen to generate the `aeson` and `lenses` stuff but it would be quite a lot more complex. We could drop `lens` support. I personally don't use it but because the API is so deep and nested I thought it might be helpful.

**Comment by loewenstein** [comment]
@aaronlevin I do like lenses though. 

My current concern is rooted in the fact that I am going to put the burden of compilation onto users (at least partially). 

Maybe I should live with the current situation until I find some time helping out.

Would you mind a PR taking out client and seev

**Comment by aaronlevin** [comment]
@loewenstein yeah, definitely OK with a PR for that. Thanks so much, I appreciate it. I haven't had much time recently to focus on this project, so this is really helpful. I think the datatypes are the most helpful part of this project.

PS - if removing the TH doesn't work, then the fundamental problem will be typeclass search in GHC and servant, which will be a real challenge to improve, but could also provide some pathological examples for the GHC team :)

**Comment by loewenstein** [comment]
@aaronlevin do you happen to have some useful links on how to profile (as
in finding possible culprits without diving into too much compiler details)
ghc?

Aaron Levin <notifications@github.com> schrieb am Mo., 21. Nov. 2016 22:20:

**Comment by aaronlevin** [comment]
@loewenstein good question. I don't. I remember I had to tweak some settings to get this to compile in the first place (related to some tics parameter that has to do with typeclass search depth or something), but no idea how to profile. The GHC devs might have some documentation somewhere.
